homeassistant/components/wallbox/config_flow.py

```diff
@@ -47,13 +70,24 @@ async def async_step_user(self, user_input=None):
         errors = {}
 
         try:
-            info = await validate_input(self.hass, user_input)
+            await self.async_set_unique_id(user_input["station"])
+            if not self._reauth_entry:
+                self._abort_if_unique_id_configured()
+                info = await validate_input(self.hass, user_input)
+
+            if not self._reauth_entry:
+                return self.async_create_entry(title=info["title"], data=user_input)
+            self.hass.config_entries.async_update_entry(
```

Comment:
Abort if the station being reauthed is not the same account that was previous authorized. Reauth should never change the underlying config entry unique id

Reply:
I did this by checking `if user_input["station"] == self._reauth_entry.data[CONF_STATION]:` and doing a `errors["base"] = "reauth_invalid"` as an else. I don't know if there is a better / standard approach to this.